Sources/Linq2Acad/AcadDatabase.cs

```diff
@@ -10,6 +10,57 @@
 
 namespace Linq2Acad
 {
+  /// <summary>
+  /// DWG version enum.
+  /// </summary>
+  public enum SaveAsDwgVersion
+  {
+    None,
+    DWG2004,
+    DWG2007,
+    DWG2010,
+    DWG2013,
+#if AutoCAD_2018 || AutoCAD_2019 || AutoCAD_2020 || AutoCAD_2021
+    DWG2018,
+#endif
+    DontChange, // Better than Current? Or Keep? EDIT: We can use Database.LastSavedAsVersion for that
+    NewestAvailable // We could remove this one, since the newest available is AutoCad20xx wit hthe higehst number
+  }
+
+  public class CreateOptions
+  {
+    public bool SaveFileOnCommit { get; set; }
```

Comment:
We could (or should?) remove `SaveFileOnCommit` here as well.

`SaveFileOnCommit` (A) and `SaveAsFileName` (B) always go together. So if A is true, B has to have a file path. If A is false, B has to be null. The independence of the two properties could provoke the situation that the user sets a file path to B, but does not set A to true, so the file is not set.

So we could remove `SaveFileOnCommit` and say if `SaveAsFileName` is set, we save on commit, otherwise we don't. Or is this an invalid implicit assumption?

Reply:
Yes please! 😉 It totally makes sense to use only the file name here, it removes the ambiguous situation (not saving but providing a filename) and it is a legit assumption. 

One can even argue that we should `keepDatabaseOpen` if no file name is given (but we can't assume anything if it is)...

Reply:
`keepDatabaseOpen` is really an advanced option and should be used with caution. It's better to not touch it from within API.

BTW I once had a use case for creating an in-memory database without saving it. An Autodesk developer somewhere provided a great sample that contains a WinForms Control that hosts and displays an AutoCAD drawing database. I used the in-memory database to display Entities that were generated on-the-fly and displayed, but never saved.